sql/core/src/test/resources/sql-tests/inputs/predicate-functions.sql

```diff
@@ -0,0 +1,42 @@
+-- In
```

Comment:
Could we move these to `PredicateSuite` and then we can use `checkEvaluation`?

Reply:
Actually I feel the test coverage might be out of the scope of this PR a little bit. This PR is for the expression description. If you don't insist we add test here together, I'd like to submit other PR to improve the test coverage.

Reply:
If we do not have a test case, how can we know the description is correct?

Reply:
Let me explain it a little bit more. No need to add more and more test cases in this PR. We need to ensure anything claimed in the descriptions is well tested; if no test case exists, we need to add it into this PR. It should include both negative and positive test cases. For example, map is not supported; and then this needs to be documented and tested.

Reply:
The test cases should be added along with the actual code changes such as #18818 that changes the underlying semantics.

For example, I saw there are tests added to test map is not supported by #18818: https://github.com/apache/spark/blob/cba69aeb453d2489830f3e6e0473a64dee81989e/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ExpressionTypeCheckingSuite.scala#L112-L122

More tests are always good and not harm, I agreed. If the necessary tests are not added before, we should add and improve them.